### PR TITLE
chore: remove deprecated io/ioutil methods

### PR DESCRIPTION
**Pull request by notJoon**
## Description

Replace depreacated `io/ioutil` to `os` methods

### CODE REVIEW

**Review by moul — APPROVED** [review]

**Review by thehowl — APPROVED** [review]
looks good :+1: 

there's more ioutil calls throughout the codebase. I think we should also remove the `io/ioutil` in Gno's stdlibs.

do you want to tackle this here or in a separate PR?

thanks!

**Comment by notJoon** [comment]
> do you want to tackle this here or in a separate PR?

Thanks! I think it would be better to separate the PR . I'll do soon as possible 👍

File: gnovm/stdlibs/io/multi_test.gno [hunk 1]
```diff
@@ -1,9 +1,9 @@
+package io
```
Comment (thehowl) [inline gnovm/stdlibs/io/multi_test.gno:1]: actually supposed to be io_test

File: gnovm/stdlibs/io/example_test.gno [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-package io_test
+package io
```
Comment (thehowl) [inline gnovm/stdlibs/io/example_test.gno:5]: here too, io_test is correct.

File: gnovm/stdlibs/internal/itoa/itoa.gno
Comment (thehowl) [inline gnovm/stdlibs/internal/itoa/itoa.gno:1]: I don't think we need this package

File: gnovm/stdlibs/io/io_test.gno [hunk 3]
```diff
@@ -1,9 +1,9 @@
+package io
```
Comment (thehowl) [inline gnovm/stdlibs/io/io_test.gno:1]: supposed to be io_test

File: gnovm/stdlibs/io/io_test.gno [hunk 4]
```diff
@@ -434,28 +434,111 @@ func TestSectionReader_Size(t *testing.T) {
...
-
-	want = errors.New("largeWriterError")
-	rb = new(Buffer)
-	wb = largeWriter{err: want}
-	rb.WriteString("hello, world.")
-	if _, err := io.Copy(wb, rb); err != want {
-		t.Errorf("Copy error: got %v, want %v", err, want)
-	}
-}
+// type largeWriter struct {
```
Comment (thehowl) [inline gnovm/stdlibs/io/io_test.gno:437]: why?

File: gnovm/stdlibs/io/io.gno [hunk 5]
```diff
@@ -543,6 +554,53 @@ func (s *SectionReader) ReadAt(p []byte, off int64) (n int, err error) {
 // Size returns the size of the section in bytes.
 func (s *SectionReader) Size() int64 { return s.limit - s.base }
 
+// And OffsetWriter maps writers at offset base to offset base + off in the underlying writer.
```
Comment (thehowl) [inline gnovm/stdlibs/io/io.gno:557]:
```suggestion
// An OffsetWriter maps writers at offset base to offset base + off in the underlying writer.
```

**Review by thehowl — CHANGES_REQUESTED** [review]

File: gnovm/stdlibs/io/io_test.gno (hunk 3 above)
Comment (notJoon) [reply gnovm/stdlibs/io/io_test.gno:1]: Adding the `_test` suffix to packages causes a `... already exists in cache` panic, so I temporarily removed it, but forgot to change it.

File: gnovm/stdlibs/io/io_test.gno (hunk 4 above)
Comment (notJoon) [reply gnovm/stdlibs/io/io_test.gno:437]: WIP. I've commented it out because it's not working properly yet. I'll get to it as soon as I can.

**Comment by notJoon** [comment]
@thehowl Thanks for your suggest! I think now I'm done with the `ioutil` stuff now. I skipped the documents because it appears to be updated in #1311.

However, I'm still getting tests failing on the action, but when I run it locally, it works fine. I'm not sure what to do about that.

**Review by thehowl — APPROVED** [review]

**Comment by thehowl** [comment]
this looks good, I asked for another review just to be sure but we should be good to go.

**Review by moul — APPROVED** [review]

**Pull request merged**